src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -119,58 +85,96 @@ protected function mapPluginsByResourceName(): void
      */
     protected function getEffectivePlugins(array $resources): array
     {
-        $effectivePlugins = [];
         if (empty($resources)) {
             return $this->eventResourcePlugins;
         }
 
+        $filteredPlugins = array_filter($this->eventResourcePlugins, function (EventResourcePluginInterface $eventResourcePlugin) use ($resources) {
+            return in_array($eventResourcePlugin->getResourceName(), $resources);
+        });
+
         foreach ($resources as $resource) {
-            if (!isset($this->eventResourcePlugins[$resource])) {
+            if (!array_filter($filteredPlugins, function (EventResourcePluginInterface $eventResourcePlugin) use ($resource) {
```

Comment:
Preferable to not use such complex things inside the IF statement

Reply:
Extracted variable to fix this